util/ribbon_impl.h

```diff
@@ -126,15 +122,31 @@ struct AddInputSelector<Key, ResultRow, true /*IsFilter*/> {
 // can do" with available hash information in terms of FP rate and
 // compactness. (64 bits recommended and sufficient for PHSF practical
 // purposes.)
+//
+// Another feature of this hasher is a minimal "premixing" of seeds before
+// they are provided to TypesAndSettings::HashFn in case that function does
+// not provide sufficiently independent hashes when iterating merely
+// sequentially on seeds. (This for example works around a problem with the
+// preview version 0.7.2 of XXH3 used in RocksDB, a.k.a. XXH3p or Hash64, and
```

Comment:
Why not just use the new version of XXH3 for ribbon? to avoid this next seed calculation.

Reply:
We would then need to maintain two versions of XXH code in RocksDB, which not only increases maintenance complexity, but also working code size when using both versions.

But even that concern is based on an assumption I expect to be false: I expect only to use StandardRehasher in RocksDB, and only use one seed (default, like Bloom filter) for hashing to initial 64 bits.

So this fix for handling seeds was really about validating and testing the implementation's generality than it was about what we expect to use in RocksDB. (I might have even deceived myself for a while on that point.)

Reply:
> So this fix for handling seeds was really about validating and testing the implementation's generality than it was about what we expect to use in RocksDB. (I might have even deceived myself for a while on that point.)

Correction: StandardRehasher needed to do some kind of mixing on "ordinal seeds" anyway for it to work properly, and doing that ahead of time saves maybe a few per-query cycles. The arguably flawed handling of seeds by our built-in hash functions just made it an obvious choice to share the seed pre-mixing infrastructure for both use-cases (sufficiently rehashing/re-seeding a precomputed hash, or independently seeding a stock hash function).